lnrpc/routerrpc/router_backend.go

```diff
@@ -45,6 +46,8 @@ type RouterBackend struct {
 	// that we receive payment requests that send to destinations on our
 	// network.
 	ActiveNetParams *chaincfg.Params
+
+	Tower channeldb.ControlTower
```

Comment:
If we only need payment subscription, then we can use a much slimmer interface here: an interface with only `SubscribePayment`.

Reply:
I prefer to only add those interfaces when we actually need them.

Reply:
Rephrased the router backend now depends on a number of methods that it doesn't actually need. Any thing that mocks this `Tower` variable now needs to unnecessarily create a bunch of empty methods, when we only need a method that exposes the subscription functionality.

Reply:
Yes, I understand that in general, but I mean that we currently don't mock that `Tower` variable. I'd like to postpone creating that interface until we need to mock that. But if it is a blocker for you, let me know. I don't mind adding it.